docs/dinamicko-programiranje/knapsack.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Knapsack
+---
+
+### Problem
+
+Ruksak kapaciteta $x$ moramo napuniti sa stvarima kojih ima $n$, a od kojih svaka zauzima određeni kapacitet i ima određenu vrijednost. Naravno cilj nam je maksimizirati ukupnu vrijednost stvari u ruksaku. U zadatku bitnu razliku čini jesu li stvari tekuće ili čvrste.
+
+### Rješenje
+
+Ako su stvari tekuće onda pohlepno možemo uzimati stvari što veće vrijednosti ili što većeg omjera količine i vrijednosti ovisno o tome kako je zadan zadatak.
+
+Ako su stvari čvrste tada nam isti pohlepni pristup neće dati točno rješenje i to je jednostavno pokazati. Neka je kapacitet ruksaka $10$ i imamo $3$ stvari oblika $\{value, size\}$: $\{5, 9\}, \{4, 4\}, \{4, 5\}$. Pohlepni pristup uzeo bi stvar vrijednosti $5$ i više ništa ne bi stalo u ruksak, a bolja opcija je uzeti ostale dvije stvari i postići vrijednost $8$.
```

Comment:
Pohlepni pristup bi uzeo prvo (4,4) pa (4,5) i dao točno rješenje.
Situacija u kojoj ne bi je npr (7,8), (4,5), (4,5) (kapacitet 10)

Reply:
Popravio